hotel/src/hil/rng.rs

```diff
@@ -0,0 +1,25 @@
+/// A Random Number Generator, based on the Rng trait from the rand crate.
+pub trait Rng {
+    /// Returns 4 bytes of random data.  Types that implement the Rng trait
+    /// only need to implement this function.
+    fn next_u32(&mut self) -> u32;
```

Comment:
On the surface, it seems to me that this should be split phase. A random number can only be generated every so-many cycles, so eventually you'll have to block. I don't know about hotel specifically, but something like 32-bits every 80 cycles seems normal, which means generating a 1024-bit random number would block for 32 microseconds or so ~200uS...